lib/galaxy/tools/cache.py

```diff
@@ -69,49 +80,54 @@ def _should_cleanup(self, config_filename):
 
     def get_tool(self, config_filename):
         """Get the tool at `config_filename` from the cache if the tool is up to date."""
-        return self._tools_by_path.get(config_filename, None)
+        with self._lock:
```

Comment:
I don't think this is necessary or a good idea, getting things from a dict is an atomic operation. I think we only need to lock add and remove (and I think that's why cleanup was already wrapped in try/except).